<?xml version="1.0"?>
<launch>
	<arg name="record" default="false"/>
    <arg name="duration" default="360"/>
    <arg name="rec_name" default="multidrone.bag"/>

	<env name="PX4_HOME_LAT" value="19.134423"/>
	<env name="PX4_HOME_LON" value="72.911763"/>

	<include file="$(find task_4)/launch/multi_uav_mavros_sitl_sdf.launch"/>

	<node name="throttler_model_states" type="throttle" pkg="topic_tools" args="messages /gazebo/model_states 10"/>
    <node name="edrone_1_gripper_service" type="edrone1_gripper_service.py" pkg="gazebo_ros_link_attacher"/>
    <node name="edrone_2_gripper_service" type="edrone2_gripper_service.py" pkg="gazebo_ros_link_attacher"/>
    <!--
       -<node name='multidrone' type='multidrone.py' pkg='task_4' />
       -->
	<group if="$(arg record)">
		<node name="rosbag_record_multidrone" pkg="rosbag" type="record"
            args="record -O $(find task_4)/bag_files/$(arg rec_name) --duration=$(arg duration) --chunksize=10 /edrone0/mavros/setpoint_position/local /edrone1/mavros/setpoint_position/local  /edrone0/mavros/local_position/pose /edrone1/mavros/local_position/pose /gazebo/model_states_throttle " output="screen"/>

	</group>

</launch>
